experimental/iterators/.clang-format

```diff
@@ -0,0 +1,2 @@
+---
+BasedOnStyle: Google
```

Comment:
Note, we generally follow LLVM's coding style: https://llvm.org/docs/CodingStandards.html

While things are under experimental, no need to change.

Reply:
OK, sure, if the code lives here, then the LLVM style makes sense.

Reply:
The earlier fix this, the cheaper --> changed with the latest commit.

Reply:
snakeCasing is only applied in a few places at this point though